src/review-todo.js

```diff
@@ -0,0 +1,212 @@
+let isDate = require("date-fns/isDate");
+
+function todo(todoName = "New Todo") {
+	let title = todoName,
+		description = "",
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList();
+
+	let todoObj = Object.create(todo.proto);
+	Object.assign(todoObj, {
+		title,
+		description,
+		isDone,
+		urgency,
+		dueDate,
+		scheduleDate,
+		tags,
+	});
+
+	return todoObj;
+}
+
+todo.proto = {};
+
+function project(projectName = "New Project") {
+	let proto = Object.assign(project.proto, { isChild });
+	let projectObj = Object.create(proto);
+
+	let title = projectName,
+		description = "",
+		favorite = false,
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList(),
+		_todoList = createTodoList(projectObj),
+		todoList = Object.create(_todoList),
+		createTag = tagGenerator("project").bind(projectObj);
+
+	todoList.add = (proj) => {
+		if (
+			Object.getPrototypeOf(proj) === project.proto &&
+			projectObj.isChild(proj)
+		) {
+			throw "Looping projects";
+		}
+		_todoList.add(proj);
+	};
+
+	function isChild(project) {
+		let projects = this.tags
+			.list()
+			.filter((tag) => tag.tagType === "project");
+		if (!projects) return false;
+		if (projects.some((proj) => proj.id === project)) return true;
```

Comment:
Where is this `id` property coming from?

Reply:
If a Todo item is part of a Project it also has that Project's tag (the idea was to have one Todo be able to be a part of many Projects, or to have todo lists for different contacts that might share items).  So tags had the type of object and the id which was the object itself, I've been regretting the choice of name.

I might be able to use something like instanceof instead of making tags myself but the same article that pushed me away from constructor functions was actually using the inconsistencies of the instanceof function as an argument and I was hesitant to do any kind of prototype checking (though I do use it when I check for looping Projects).

I'll look into it more, I want to practice with prototypes.

Reply:
I think giving them some type of `id` is a great idea! I just don't see an `id` field defined on `project` objects. I wasn't sure if their automatic, inherited, or not defined yet.